tests/math_tools/test_partial_derivatives.cpp

```diff
@@ -15,164 +17,439 @@ struct X
 {
     static bool constexpr PERIODIC = false;
 };
-using CoordX = Coord<X>;
-struct BSplinesX : ddc::UniformBSplines<X, 3>
+
+struct Y
 {
+    static bool constexpr PERIODIC = false;
 };
-auto constexpr SplineXBoundary = ddc::BoundCond::GREVILLE;
-using SplineInterpPointsX
-        = ddc::GrevilleInterpolationPoints<BSplinesX, SplineXBoundary, SplineXBoundary>;
 
-struct GridX : NonUniformGridBase<X>
+/**
+ * @brief A class that represents a polynomial test function for computing partial derivatives.
+ * The polynomial depends on two variables.
+ */
+template <class DerivativeDimension, std::size_t spline_degree>
+class FunctionToDifferentiatePolynomial
 {
+    static_assert(std::is_same_v<DerivativeDimension, X> || std::is_same_v<DerivativeDimension, Y>);
+
+    using DDim = DerivativeDimension;
+    using ODim = std::conditional_t<std::is_same_v<DDim, X>, Y, X>;
+
+    using CoordFull = Coord<DDim, ODim>;
+
+    int const m_degree;
+
+public:
+    FunctionToDifferentiatePolynomial() : m_degree(spline_degree + 1) {}
+    /**
+     * @brief Get the value of the function at given coordinate.
+     *
+     * @param[in] coord_xy The coordinate where we want to evaluate
+     * the function.
+     *
+     * @return The value of the function at the coordinate.
+     */
+    KOKKOS_FUNCTION double operator()(CoordFull const coord) const
+    {
+        double const x = ddc::get<X>(coord);
+        double const y = ddc::get<Y>(coord);
+        return (ipow(x, m_degree + 1) + ipow(y, m_degree)) * y;
+    }
+
```

Comment:
We may want to consider a little more randomness. E.g. see:
https://github.com/CExA-project/ddc/blob/main/tests/splines/polynomial_evaluator.hpp

Reply:
I am ok with this though